File: src/main/java/ru/gdlbo/search/searcher/config/GlobalExceptionHandler.java
package ru.gdlbo.search.searcher.config;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.MissingServletRequestParameterException;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

import java.util.HashMap;
import java.util.Map;

@ControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(MissingServletRequestParameterException.class)
    public ResponseEntity<Map<String, String>> handleMissingServletRequestParameterException(MissingServletRequestParameterException ex) {
        Map<String, String> response = new HashMap<>();
        response.put("message", "Required request parameter '" + ex.getParameterName() + "' is not present");
        return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
    }
}

File: src/main/java/ru/gdlbo/search/searcher/config/GlobalExceptionHandler.kt
package ru.gdlbo.search.searcher.config

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.MissingServletRequestParameterException
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler

@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(MissingServletRequestParameterException::class)
    fun handleMissingServletRequestParameterException(ex: MissingServletRequestParameterException): ResponseEntity<Map<String, String>> {
        val response: MutableMap<String, String> = HashMap()
        response["message"] = "Required request parameter '" + ex.parameterName + "' is not present"
        return ResponseEntity(response, HttpStatus.BAD_REQUEST)
    }
}